support/cas-server-support-ldap-core/src/main/java/org/apereo/cas/authentication/support/DefaultAccountStateHandler.java

```diff
@@ -156,14 +156,16 @@ protected void handleWarning(
             return;
         }
 
-        final ZonedDateTime expDate = DateTimeUtils.zonedDateTimeOf(warning.getExpiration());
-        final long ttl = ZonedDateTime.now(ZoneOffset.UTC).until(expDate, ChronoUnit.DAYS);
-        LOGGER.debug(
+        if(warning.getExpiration()!=null){
```

Comment:
You are right, I will add a log message in case of an already expired password. When password is not expired, the log message already exists.
Do you know if there is a case that getLoginRemains is greater than 0 while password hasn't expired yet? In OpenLDAP, I think that there is no such case, but I don't know for other LDAP Servers...
If other LDAP servers, behave like OpenLDAP then these two if statements, should become one if - else...